android/ouinet/src/main/java/ie/equalit/ouinet/ConnectivityBroadcastReceiver.kt

```diff
@@ -0,0 +1,47 @@
+package ie.equalit.ouinet
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+import android.util.Log
+
+class ConnectivityBroadcastReceiver (
+    background: OuinetBackground
+    ) : BroadcastReceiver() {
+
+    private var background : OuinetBackground? = null
+
+    init {
+        this.background = background
+    }
+
+    override fun onReceive(context: Context?, intent: Intent) {
+        val info = intent.getParcelableExtra<NetworkInfo>(ConnectivityManager.EXTRA_NETWORK_INFO)
+            ?: return
+
+        // Restart the Ouinet client whenever connectivity has changed and become stable.
```

Comment:
This comment is useful but I think code block below could be encapsulated in a method, e.g.  `restartOnConnectivityChange(..)`

Reply:
Good suggestion! If something needs a comment, just put in a method :slightly_smiling_face:. I didn't refactor this ConnectivityBroadcastReceiver code very much because it has a ton of deprecation warnings, so it might need a complete overhaul sooner rather than later.